src/args.rs

```diff
@@ -312,6 +312,48 @@ impl Args {
     }
 }
 
+// This macro is used to generate a member function of `ArgMatches` which
+// parses an argument item of the form `[0-9]*(KMG)?`.
+//
+// This is a macro instead of a generic function due to difficulties in
+// correctly constraining the output $ty to work for a number of integers.
+// Since there are limited numbers of types that currently need this function
+// we perform the type monomorphization ourselves.
+macro_rules! human_readable_size_arg_impl {
```

Comment:
I'm not sure this macro is necessary, and I'd really like to find a way to get rid of it. Can it be written to specifically return a `u64` and then just convert the result to a `usize`? (That conversion should return an error if the `u64` is greater than [`std::usize::MAX`](https://doc.rust-lang.org/std/usize/constant.MAX.html).)